src/jit/flowgraph.cpp

```diff
@@ -438,6 +438,12 @@ void Compiler::fgEnsureFirstBBisScratch()
         {
             block->inheritWeight(fgFirstBB);
         }
+
+        // The first basic block will have a ref count of at least one. This corresponds to the virtual edge
+        // from the prolog to the entry block. We need to remove this virtual edge before adding the scratch
+        // block as a predecessor.
+        fgFirstBB->bbRefs--;
+        fgAddRefPred(fgFirstBB, block);
```

Comment:
Maybe you should assert that bbRefs > 0?

You could also avoid making this change by calling fgEnsureFirstBBisScratch() during fgMorphCall if we see such a call, in preparation for having it available later.